lib/embulk/guess/schema_guess.rb

```diff
@@ -62,11 +62,8 @@ def guess_type(str)
           return "long"
         end
 
-        if str.include?('.')
-          a, b = str.split(".", 2)
-          if a.to_i.to_s == a && b.to_i.to_s == b
-            return "double"
-          end
+        if str =~ /^[-+]?([1-9]\d*|0)(\.\d+)([eE][-+]?[0-9]+)?$/
```

Comment:
Fixed https://github.com/embulk/embulk/pull/554/commits/9a2548a9e9d85db77407dbdf3d3482488e653746 How about this?